Vungle/VungleAdapter/ALVungleMediationAdapter.m

```diff
@@ -644,277 +437,246 @@ + (MAAdapterError *)toMaxError:(nullable NSError *)vungleError
+    [self.rewardedAdDelegate didFailToLoadRewardedAdWithError: adapterError];
 }
 
-- (void)vungleAdViewedForPlacement:(NSString *)placementID
+- (void)rewardedAdDidPresent:(VungleRewarded * _Nonnull)rewarded
 {
-    [self log: @"Ad viewed"];
-    
-    // Passing extra info such as creative id supported in 6.15.0+
-    NSString *creativeIdentifier = self.creativeIdentifiers[placementID];
-    if ( ALSdk.versionCode >= 6150000 && [creativeIdentifier al_isValidString] )
-    {
-        [self performSelector: @selector(didDisplayAdForPlacementIdentifier:withExtraInfo:)
-                   withObject: placementID
-                   withObject: @{@"creative_id" : creativeIdentifier}];
-        [self.creativeIdentifiers removeObjectForKey: placementID];
-    }
-    else
-    {
-        [self didDisplayAdForPlacementIdentifier: placementID];
-    }
-    
-    [self didStartRewardedVideoForPlacementIdentifier: placementID];
+    [self.rewardedAdDelegate didDisplayRewardedAd];
+    [self.rewardedAdDelegate didStartRewardedAdVideo];
 }
 
-- (void)vungleTrackClickForPlacementID:(NSString *)placementID
+- (void)rewardedAdDidFailToPresent:(VungleRewarded * _Nonnull)rewarded withError:(NSError * _Nonnull)withError
 {
-    [self log: @"Ad clicked"];
-    [self didClickAdForPlacementIdentifier: placementID];
+    MAAdapterError *adapterError = [MAAdapterError errorWithCode: kALErrorCode errorString: @"Ad Display Failed" mediatedNetworkErrorCode: withError.code mediatedNetworkErrorMessage: withError.localizedDescription];
+    [self.parentAdapter log: @"Rewarded ad failed to display with error: %@", adapterError];
+    [self.rewardedAdDelegate didFailToDisplayRewardedAdWithError:adapterError];
 }
 
-- (void)vungleRewardUserForPlacementID:(NSString *)placementID
+- (void)rewardedAdDidClose:(VungleRewarded * _Nonnull)rewarded
 {
-    [self log: @"Rewarded ad user did earn reward"];
-    self.grantedReward = YES;
+    [self.rewardedAdDelegate didHideRewardedAd];
 }
 
-- (void)vungleWillCloseAdForPlacementID:(NSString *)placementID
+- (void)rewardedAdDidClick:(VungleRewarded * _Nonnull)rewarded
 {
-    [self log: @"Ad will close"];
-    
-    [self didCompleteRewardedVideoForPlacementIdentifier: placementID];
-    
-    if ( self.grantedReward || [self shouldAlwaysRewardUserForPlacementIdentifier: placementID] )
-    {
-        [self didRewardUserForPlacementIdentifier: placementID withReward: [self rewardForPlacementIdentifier: placementID]];
-        self.grantedReward = NO;
-    }
+    [self.rewardedAdDelegate didClickRewardedAd];
 }
 
-- (void)vungleDidCloseAdForPlacementID:(NSString *)placementID
+- (void)rewardedAdDidRewardUser:(VungleRewarded * _Nonnull)rewarded
 {
-    [self log: @"Ad did close"];
-    [self didHideAdForPlacementIdentifier: placementID];
+    MAReward *reward = [self.parentAdapter reward];
+    [self.parentAdapter log: @"Rewarded user with reward: %@", reward];
+    [self.rewardedAdDelegate didRewardUserWithReward: reward];
```

Comment:
Might we need to set `self.grantedReward = YES;` in `rewardedAdDidRewardUser:` method ? I saw the grantedReward property is not set elsewhere. Thanks a lot. :)

Reply:
we dont need this. i am gonna remove that bool.